data/helpers.d/setting

```diff
@@ -260,45 +266,49 @@ ynh_permission_url() {
 
 # Update a permission for the app
 #
-# usage: ynh_permission_update --permission "permission" --add "group" ["group" ...] --remove  "group" ["group" ...]
-# | arg: permission - the name for the permission (by default a permission named "main" already exist)
-# | arg: add - the list of group or users to enable add to the permission
-# | arg: remove - the list of group or users to remove from the permission
+# example: ynh_permission_update --permission admin --add=samdoe --remove=all_users
+#
+# usage: ynh_permission_update --permission="permission" [--add="group1 group2"] [--remove="group1 group2"]
+# | arg: -p, --permission=  - the name for the permission (by default a permission named "main" already exist)
+# | arg: -a, --add=         - the list of group or users to enable add to the permission
+# | arg: -r, --remove=      - the list of group or users to remove from the permission
 #
-# example: ynh_permission_update --permission admin --add samdoe --remove all_users
 # Requires YunoHost version 3.7.0 or higher.
 ynh_permission_update() {
-	# Declare an array to define the options of this helper.
-	local legacy_args=par
-    declare -Ar args_array=( [p]=permission= [a]=add= [r]=remove= )
+    # Declare an array to define the options of this helper.
+    local legacy_args=par
+    local -A args_array=( [p]=permission= [a]=add= [r]=remove= )
     local permission
     local add
     local remove
     ynh_handle_getopts_args "$@"
+    add=${add:-}
+    remove=${remove:-}
 
-    if [[ -n ${add:-} ]]; then
+    if [[ -n $add ]]; then
         add="--add ${add//';'/" "}"
     fi
-    if [[ -n ${remove:-} ]]; then
+    if [[ -n $remove ]]; then
         remove="--remove ${remove//';'/" "} "
     fi
 
-    yunohost user permission update "$app.$permission" ${add:-} ${remove:-}
+    yunohost user permission update "$app.$permission" $add $remove
```

Comment:
I'm not sure, but if we don't define $add or $remove, we'll have an error here?

Edit: I've tested, no it doesn't trigger an error because add (and remove) are defined line 285 (and 286), potentially empty, but defined. I'll leave my comment here if it helps someone.

Reply:
Exactly what I was about to explain.
It fails if the variable is not set, not if it's empty.